src/main/java/com/khutircraftubackend/product/ProductService.java

```diff
@@ -24,134 +27,181 @@
-            existingProduct.setImageUrl(imageUrl);
-        }
-        return productRepository.save(existingProduct);
-    }
-
-    @Transactional
-    public void deleteProduct(Long productId) throws IOException {
-        ProductEntity existingProduct = findProductById(productId);
-        if (existingProduct.getThumbnailImageUrl() != null) {
-            String publicId = extractPublicId(existingProduct.getThumbnailImageUrl());
-            fileUploadService.deleteCloudinaryById(publicId);
-        }
-        if (existingProduct.getImageUrl() != null) {
-            String publicId = extractPublicId(existingProduct.getImageUrl());
-            fileUploadService.deleteCloudinaryById(publicId);
-        }
-        productRepository.delete(existingProduct);
-    }
-
-    @Transactional
-    public void deleteAllProductsForCurrentSeller() throws IOException {
-        SellerEntity currentSeller = sellerService.getCurrentSeller();
-        List<ProductEntity> products = productRepository.findAllBySeller(currentSeller);
-
-        for (ProductEntity product : products) {
-            if (product.getThumbnailImageUrl() != null) {
-                String publicId = extractPublicId(product.getThumbnailImageUrl());
-                fileUploadService.deleteCloudinaryById(publicId);
-            }
-            if (product.getImageUrl() != null) {
-                String publicId = extractPublicId(product.getImageUrl());
-                fileUploadService.deleteCloudinaryById(publicId);
-            }
-        }
-        productRepository.deleteBySeller(currentSeller);
-    }
-
-    private String extractPublicId(String url) {
-        String[] parts = url.split("/");
-        return parts[parts.length - 1].split("\\.")[0];
-    }
-
-    public List<ProductEntity> getProducts(int offset, int limit) {
-        Pageable pageable = PageRequest.of(offset, limit);
-        return productRepository.findAllBy(pageable);
-    }
+	
+	private final ProductRepository productRepository;
+	private final SellerService sellerService;
+	private final FileUploadService fileUploadService;
+	private final FileConverterService fileConverterService;
+	private final CategoryRepository categoryRepository;
+	private final CategoryService categoryService;
+	
+	private ProductEntity findProductById(Long productId) {
+		
+		return productRepository.findProductById(productId)
+				.orElseThrow(() -> new ProductNotFoundException("Product with id " + productId + " not found"));
+	}
+	
+	public boolean canModifyProduct(Long productId) {
+		
+		ProductEntity existingProduct = findProductById(productId);
+		
+		SellerEntity currentSeller = sellerService.getCurrentSeller();
+		
+		return (existingProduct.getSeller().getCompanyName().equals(currentSeller.getCompanyName()));
+	}
+	
+	private void validateSellerCompany(SellerEntity currentSeller, SellerEntity requestSeller) throws AccessDeniedException {
+		
+		if (!currentSeller.getCompanyName().equals(requestSeller.getCompanyName())) {
```

Comment:
Як на мене, все якраз очевидно.

Reply:
those are String objects, 
we can either 
- compare them with `==` , like `if (currentSeller.getCompanyName() == requestSeller.getCompanyName())`
- or extract 2 strings from if statement
- or implement custom equals method(s) in SellerEntity, so that we could do it this way: `currentSeller.equals(requestedSeller)`

Reply:
нашел статью. может єто дело привічки каждого девелопера. https://stackoverflow.com/questions/12775082/is-there-any-reason-for-boolean-true-equalsx-in-java

Reply:
https://www.baeldung.com/java-boolean-true-primitive-vs-constant

Reply:
> * compare them with `==`

compare them with == is not a safe practice.

> * or implement custom equals method(s) in SellerEntity, so that we could do it this way: currentSeller.equals(requestedSeller)

instead, the equals method is a best practices in comparison

Reply:
ну тоде тут здається все зрозуміло,
ми хочемо порівняти два інстанса одного типу (классу) - SellerEntity.
тому найкращий спосіб - це імплементувати `SellerEntity.equals()` метод
та передивитись назву та логіку викристання 'validateSellerCompany': 
- це треба робити не тільки на `createProduct()` а в усіх методах що змінюють будь-що в продукті. 
- якщо залишаємо це в окремому методі до змінюємо назву на 'validateSeller'
- такому методу потрібен тількр один аргумент : `SellerEntity requestSeller`, а `currentSeller` можна отримати всередині

Reply:
The check works. Why make it difficult?

Reply:
coz those three bullet points above  is right way to do it :)

Reply:
Applied override method equals() for comparison of sellers.